src/runtime/typemanager.cs

```diff
@@ -164,13 +164,13 @@ internal static IntPtr CreateType(ManagedType impl, Type clrType)
             // we want to do this after the slot stuff above in case the class itself implements a slot method
             InitializeSlots(type, impl.GetType());
 
-            if (!clrType.GetInterfaces().Any(ifc => ifc == typeof(IEnumerable) || ifc == typeof(IEnumerator)))
+            if (!(clrType.IsInterface && (clrType == typeof(IEnumerable) || clrType == typeof(IEnumerator))) &&
+                !clrType.GetInterfaces().Any(ifc => ifc == typeof(IEnumerable) || ifc == typeof(IEnumerator)))

```

Comment:
Hint: you can use `typeof(I).IsAssignableFrom(type)`

Reply:
Thanks, that became a lot cleaner.